### DISCUSSION TITLE
Can the dataset option be used with files not produced by airflow.

### DISCUSSION THREAD

**Discussion by waardd** [apache/airflow#40477] [Q&A]
I followed an example of a producer and consumer of a dataset.
Producer makes a set and that triggers the consumer to use it.
Works like a charm.

My question is, can i have a consumer function on a dataset i create myself (being via filetransfer or manual or scripted) on the system?

For example:
```
from airflow import DAG, Dataset
from airflow.decorators import task

from datetime import datetime

my_file = Dataset("/tmp/myfile_i_create.txt")

with DAG(
        dag_id="consumer_2",
        schedule=[my_file],
        start_date=datetime(2022, 1, 1),
        catchup=False
):

        @task()
        def read_my_file():
                with open(my_file, "r") as f:
                        print(f.read())

        read_my_file()
```
If i create or alter the file /tmp/myfile_i_create.txt via another method then Airflow, the consumer should trigger.

**Comment by potiuk** [comment]
In Airflow 2.9 there is na API you can call to trigger "produce" event (POST) - I think this is what you are after.
